packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -0,0 +1,580 @@
+  );
+}) as PolymorphicForwardRefComponent<'div', PanelsWrapperProps>;
+if (process.env.NODE_ENV === 'development') {
+  PanelsWrapper.displayName = 'Panels.Wrapper';
+}
+
+export const PanelsWrapperContext = React.createContext<
+  | {
+      activePanelId: string | undefined;
+      setActivePanelId: React.Dispatch<React.SetStateAction<string>>;
+      /**
+       * Simpler alternative to a full history stack.
+       *
+       * ```
+       * Record<
+       *   string, // Id of a panel
+       *   {
+       *     triggerId: string, // Id of the trigger element that points to this panel
+       *     panelId: string, // Id of the panel element in which the trigger is present
+       *   }
+       * >
+       * ```
+       */
+      triggers: Record<string, TriggerMapEntry>;
+      setTriggers: React.Dispatch<
+        React.SetStateAction<Record<string, TriggerMapEntry>>
+      >;
+      changeActivePanel: (newActiveId: string) => void;
+      shouldFocus: FocusEntry;
+      setShouldFocus: React.Dispatch<React.SetStateAction<FocusEntry>>;
+      panels: React.MutableRefObject<Set<string>>;
+    }
+  | undefined
+>(undefined);
+if (process.env.NODE_ENV === 'development') {
+  PanelsWrapperContext.displayName = 'PanelsWrapperContext';
+}
+
+// #endregion PanelsWrapper
+// ----------------------------------------------------------------------------
+// #region Panel
+
+type PanelProps = {
+  id: string;
+};
+
+const Panel = React.forwardRef((props, forwardedRef) => {
+  const { id, children, className, ...rest } = props;
+
+  const { activePanelId, triggers, panels, setActivePanelId } =
+    useSafeContext(PanelsWrapperContext);
+
+  const associatedTrigger = React.useMemo(() => triggers[id], [id, triggers]);
+
+  const previousActivePanelId = useDelayed(activePanelId) || activePanelId;
+
+  const isMounted = [activePanelId, previousActivePanelId].includes(id);
+  const isTransitioning =
+    activePanelId === id && activePanelId !== previousActivePanelId;
+  const isInert = previousActivePanelId === id && activePanelId !== id;
+
+  useLayoutEffect(() => {
+    // In the beginning, show the first <Panel> in the DOM
+    const isFirstPanel = activePanelId == null && panels.current.size === 0;
+    if (isFirstPanel) {
+      // Set the first panel as active without calling onActiveIdChange since this is initialization.
+      setActivePanelId(id);
+    }
+
+    const panelsCurrent = panels.current;
+    if (!panelsCurrent.has(id)) {
+      panelsCurrent.add(id);
+    }
+
+    return () => {
+      panelsCurrent.delete(id);
+    };
+  }, [activePanelId, id, panels, setActivePanelId]);
+
+  return (
+    <PanelContext.Provider
+      value={React.useMemo(
+        () => ({ id, associatedTrigger }),
+        [associatedTrigger, id],
+      )}
+    >
+      {isMounted && (
+        <Box
+          ref={forwardedRef}
+          id={id}
+          className={cx('iui-panel', className)}
+          aria-labelledby={`${id}-header-title`}
```

Comment:
It might work in VoiceOver but it's not something we can take for granted to work everywhere, because labeling generic elements is not allowed.

The addition of `role="group"` looks good now. 👍

Reply:
Like I said, it looks good. I was just clarifying that your assumption about `role="group"` being redundant might not be universally correct. We should definitely test on other screen-readers too.

Resolving.